### ISSUE TITLE
[dropdown] [multiple] - Select order not correct.

### ISSUE THREAD

**Issue by tmikaeld** [Semantic-Org/Semantic-UI#3841]
http://jsfiddle.net/czw7zy25/

Works fine with smaller ID's like 1,2,3 but with higher numbers like these, the select order is not correct - instead it adds the values in the order that they are listed in the dropdown.

**Comment by neutraali** [comment]
This same issue applies in the situation where the dropdown input has initial values, for example:

`<input name="numbers" multiple type="hidden" value="162,16,136,129">`

The dropdown will sort them in the order that they're provided in the markup. This is a problem where, for example, I want the dropdown _choices_ to be alphabetically sorted, but I want the _selection_ (e.g. values, labels) to be in the exact order that they're provided.

**Comment by brunotourinho** [comment]
Any workarounds to this issue?

**Comment by ponvino** [comment]
Any Update or workaround for this issue?

It is blocking to implement my requirement as well, where i need to get the selected values of multi select dropdown in the same order as user selected.

Thanks,

**Comment by staff0rd** [comment]
The reason is due to [this code](https://github.com/Semantic-Org/Semantic-UI/blob/8dda85c61bacd5ec92ed703f9e472f7d5e8f178b/dist/components/dropdown.js#L1900-L1907) which sets selected by iterating through the declared options and tests for matches against set options.

A possible fix here would be to iterate through the set options and test for matches against the declared options instead.

**Comment by jogee** [comment]
@staff0rd You are right, I've debugged the issue and the first part of the problem is that they loop the items instead of the passed values. Another issue is that the jquery add method which is used adds values in the order that they appear in the dom.

I could try to add a pull request but I'm not really familiar with the code.

**Comment by jogee** [comment]
I've tried to create a pull request but the userAdditions gets appended after the normal additions and the change would be quite intrusive.

Eventually worked around this, you can call the selected after you add the options to the dom in the order that you want them to appear and then refresh the dropdown.

**Comment by ngouy** [comment]
This bug is real. It needs a patch.

**Comment by brunotourinho** [comment]
bump!

**Comment by cashsun** [comment]
This is a bug, especially for multiple search, when user input label the label moves to arbitrary position of label list

**Comment by FichteFoll** [comment]
I'm also working on something where the order of the selection is important and to be preserved. This is fine when the list is modified while the page is loaded, but when the dropdown is first rendered with an initial value, it reorders the item and displays them lexicographically sorted, as others have mentioned.

Note that the `set selected` workaround only works if there isn't already a default value.

**Comment by huypn1106** [comment]
I worked around by set one by one value, not all in the array

**Comment by tmikaeld** [comment]
@huypn1106 Yeah, there are some workarounds that work... but it shouldn't be needed.

I'm going to switch away from semantic and fomantic, nothing is being updated and pull requests are not being merged.
